### PR TITLE
[gamepad] Avoid OOB on empty belt

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/controls/plrctrls.cpp [hunk 1]
```diff
@@ -1509,6 +1509,9 @@ void UseBeltItem(int type)
 {
 	for (int i = 0; i < MAXBELTITEMS; i++) {
 		auto &myPlayer = Players[MyPlayerId];
+		if (myPlayer.SpdList[i].isEmpty()) {
```
Comment (qndel) [inline Source/controls/plrctrls.cpp:1512]: get rid of the same check below?

Comment (AJenbo) [reply Source/controls/plrctrls.cpp]: Thanks, I also made it somewhat readable, hope you agree :)

File: Source/controls/plrctrls.cpp [hunk 2]
```diff
@@ -1508,16 +1508,21 @@ void plrctrls_after_game_logic()
...
+		if (item.isEmpty()) {
+			continue;
+		}
+
+		const item_misc_id id = AllItemsList[item.IDidx].iMiscId;
+		const spell_id spellId = AllItemsList[item.IDidx].iSpell;
+
+		bool isRejuvination = IsAnyOf(id, IMISC_REJUV, IMISC_FULLREJUV);
+		bool isHealing = isRejuvination || IsAnyOf(id, IMISC_HEAL, IMISC_FULLHEAL) || (id == IMISC_SCROLL && spellId == SPL_HEAL);
+		bool isMana = isRejuvination || IsAnyOf(id, IMISC_MANA, IMISC_FULLMANA);
```
Comment (StephenCWills) [inline Source/controls/plrctrls.cpp:1521]:
```suggestion
		bool isRejuvenation = IsAnyOf(id, IMISC_REJUV, IMISC_FULLREJUV);
		bool isHealing = isRejuvenation || IsAnyOf(id, IMISC_HEAL, IMISC_FULLHEAL) || (id == IMISC_SCROLL && spellId == SPL_HEAL);
		bool isMana = isRejuvenation || IsAnyOf(id, IMISC_MANA, IMISC_FULLMANA);
```

File: Source/controls/plrctrls.cpp [hunk 3]
```diff
@@ -1508,16 +1508,21 @@ void plrctrls_after_game_logic()
...
+		Item &item = Players[MyPlayerId].SpdList[i];
+		if (item.isEmpty()) {
+			continue;
+		}
+
+		const item_misc_id id = AllItemsList[item.IDidx].iMiscId;
+		const spell_id spellId = AllItemsList[item.IDidx].iSpell;
+
+		bool isRejuvenation = IsAnyOf(id, IMISC_REJUV, IMISC_FULLREJUV);
+		bool isHealing = isRejuvenation || IsAnyOf(id, IMISC_HEAL, IMISC_FULLHEAL) || (id == IMISC_SCROLL && spellId == SPL_HEAL);
```
Comment (obligaron) [inline Source/controls/plrctrls.cpp:1520]: If we use `item` directly, we could use some helper functions of it:
```suggestion
		bool isHealing = isRejuvenation || IsAnyOf(id, IMISC_HEAL, IMISC_FULLHEAL) || item.IsScrollOf(SPL_HEAL);
```

Comment (ephphatha) [reply Source/controls/plrctrls.cpp:1520]: `IsScrollOf` also includes targeted scrolls, assuming that they all use the correct spell id of `SPL_HEALOTHER` though so this should be functionally the same.

**Comment by AJenbo** [comment]
This crash is proving the hardest to solve, but not for anything related to the bug it self :D

File: Source/controls/plrctrls.cpp (hunk 3 above)
Comment (AJenbo) [reply Source/controls/plrctrls.cpp]: Applied

File: Source/controls/plrctrls.cpp [hunk 4]
```diff
@@ -1508,16 +1508,21 @@ void plrctrls_after_game_logic()
...
-				UseInvItem(MyPlayerId, INVITEM_BELT_FIRST + i);
-				break;
-			}
+		Item &item = Players[MyPlayerId].SpdList[i];
+		if (item.isEmpty()) {
+			continue;
+		}
+
+		const item_misc_id id = AllItemsList[item.IDidx].iMiscId;
+		const spell_id spellId = AllItemsList[item.IDidx].iSpell;
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply Source/controls/plrctrls.cpp]: Done

File: Source/controls/plrctrls.cpp [hunk 5]
```diff
@@ -1508,16 +1508,18 @@ void plrctrls_after_game_logic()
...
-			if (!myPlayer.SpdList[i].isEmpty()) {
-				UseInvItem(MyPlayerId, INVITEM_BELT_FIRST + i);
-				break;
-			}
+		Item &item = Players[MyPlayerId].SpdList[i];
+		if (item.isEmpty()) {
+			continue;
+		}
+
+		bool isRejuvination = IsAnyOf(item._iMiscId, IMISC_REJUV, IMISC_FULLREJUV);
```
Comment (qndel) [inline Source/controls/plrctrls.cpp:1516]:
```suggestion
		bool isRejuvenation = IsAnyOf(item._iMiscId, IMISC_REJUV, IMISC_FULLREJUV);

```

**Review by StephenCWills — APPROVED** [review]
It looks much nicer to me.

**Comment by AJenbo** [comment]
Not to mention it won't crash the game anymore :)

**Pull request merged**